src/Core/Parser/Scene.cpp

```diff
@@ -145,6 +163,15 @@ int Raytracer::Scene::_parseLightsSetting(const libconfig::Setting &lights)
 
             newPoint->setPosition(position);
             newPoint->setDiffuseMultiplier(diffuse);
+
+            if (lightArrayPoint[index].exists("translation")) {
+                libconfig::Setting& translation = lightArrayPoint[index].lookup("translation");
+                Math::Vector3D trans(_parseValue(translation["x"]), _parseValue(translation["y"]), _parseValue(translation["z"]));
+                Math::Vector3D neworigin = newPoint->getPosition();
+                neworigin.translate(trans);
+                newPoint->setPosition(neworigin);
```

Comment:
Rename the variable to newOrigin.

```suggestion
                Math::Vector3D newOrigin = newPoint->getPosition();
                newOrigin.translate(trans);
                newPoint->setPosition(newOrigin);
```